resource/modify_plan.go

```diff
@@ -26,6 +27,9 @@ type ModifyPlanRequest struct {
 
 	// ProviderMeta is metadata from the provider_meta block of the module.
 	ProviderMeta tfsdk.Config
+
+	// Private is provider private state data.
```

Comment:
Nit: Maybe we should say this is "resource" private state data, mention it will pre-populate the response data, and how to work with it? `ReadRequest`/`ReadResponse` has some expanded verbiage we can borrow.